prow/moonraker/moonraker.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package moonraker
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+)
+
+const (
+	PathGetInrepoconfig = "inrepoconfig"
+)
+
+type Moonraker struct {
+	ConfigAgent       *config.Agent
+	InRepoConfigCache *config.InRepoConfigCache
+}
+
+type ProwYAMLGetter interface {
+	GetProwYAML(refs *prowapi.Refs) (*config.ProwYAML, error)
+}
+
+// serveGetInrepoconfig returns a ProwYAML object marshaled into JSON.
+func (mr *Moonraker) ServeGetInrepoconfig(w http.ResponseWriter, r *http.Request) {
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		logrus.WithError(err).Info("unable to read request")
+		http.Error(w, fmt.Sprintf("bad request %v", err), http.StatusBadRequest)
+		return
+	}
+
+	refs := &prowapi.Refs{}
+	err = json.Unmarshal(body, refs)
+	if err != nil {
+		logrus.WithError(err).Info("unable to unmarshal getInrepoconfig request")
+		http.Error(w, fmt.Sprintf("unable to unmarshal getInrepoconfig request: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	baseSHAGetter := func() (string, error) {
+		return refs.BaseSHA, nil
+	}
+	var headSHAGetters []func() (string, error)
+	for _, pull := range refs.Pulls {
+		pull := pull
+		headSHAGetters = append(headSHAGetters, func() (string, error) {
+			return pull.SHA, nil
+		})
+	}
+	identifier := refs.Org + "/" + refs.Repo
+
+	prowYAML, err := mr.InRepoConfigCache.GetProwYAML(identifier, baseSHAGetter, headSHAGetters...)
+	if err != nil {
+		logrus.WithError(err).Error("unable to retrieve inrepoconfig ProwYAML")
+		http.Error(w, fmt.Sprintf("unable to retrieve inrepoconfig ProwYAML: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
```

Comment:
nit: This is unnecessary since it is the default value.

Reply:
Fixed locally.